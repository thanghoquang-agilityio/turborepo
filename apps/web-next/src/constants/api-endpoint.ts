export const API_ENDPOINT = {
  PRODUCTS: 'products',
  PRODUCT_VARIANTS: 'product-variants',
  CATEGORIES: 'categories',
  CARTS: 'carts',
  USERS: 'users',
}

export const USER_ID_DEFAULT = "1"

export const API_URL =
  process.env.NEXT_PUBLIC_API_URL || 'https://api.example.com'
export const API_IMAGE_URL =
  process.env.NEXT_PUBLIC_API_IMAGE_URL || 'https://images.example.com'
