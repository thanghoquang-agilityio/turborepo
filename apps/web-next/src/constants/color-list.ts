export const COLOR_ITEMS = [
  {
    color: 'wheat',
    backgroundColor: 'bg-warning-300',
    borderColor: 'border-warning-300',
  },
  {
    color: 'yellow green',
    backgroundColor: 'bg-success-200',
    borderColor: 'border-success-200',
  },
  {
    color: 'slate blue',
    backgroundColor: 'bg-secondary-400',
    borderColor: 'border-secondary-400',
  },
  {
    color: 'violet',
    backgroundColor: 'bg-danger-200',
    borderColor: 'border-danger-200',
  },
  {
    color: 'salmon',
    backgroundColor: 'bg-danger-300',
    borderColor: 'border-danger-300',
  },
  {
    color: 'dark green',
    backgroundColor: 'bg-success-300',
    borderColor: 'border-success-300',
  },
]

export const COLOR_ITEM_DEFAULT = { backgroundColor: '', borderColor: '' }
