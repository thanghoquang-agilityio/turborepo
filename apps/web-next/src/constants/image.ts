export const IMAGE_NOT_AVAILABLE = 'UnavailableImage'
export const SRC_IMAGE_NOT_AVAILABLE = '/images/image-not-available.webp'

export const SRC_BANNER_HEADPHONE = '/images/banner-headphone.webp'

export const IMAGE_DEFAULT = {
  name: '',
  url: SRC_IMAGE_NOT_AVAILABLE,
}
