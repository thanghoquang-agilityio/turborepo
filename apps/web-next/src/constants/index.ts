export * from './api-endpoint'
export * from './color-list'
export * from './image'
export * from './message'
export * from './navbar'
export * from './pagination'
export * from './product'
export * from './regex'
export * from './relationship'
export * from './router'
export * from './styles'
