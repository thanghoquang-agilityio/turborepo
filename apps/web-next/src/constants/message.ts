export const CART_MESSAGES = {
  ADD: {
    SUCCESS: 'You have successfully added to your Cart.',
    FAILED: 'An error occurred while adding to your Cart. Please try again.',
    WARNING: 'Your cart already contains all products in stock.',
  },
  UPDATE: {
    SUCCESS: 'You have successfully updated to your Cart.',
    FAILED: 'An error occurred while updating to your Cart. Please try again.',
  },
  REMOVE: {
    SUCCESS: 'You have successfully removed from your Cart.',
    FAILED:
      'An error occurred while removing from your Cart. Please try again.',
  },
}
