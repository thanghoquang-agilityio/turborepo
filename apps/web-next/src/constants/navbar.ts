export const NAVBAR_LIST = [
  {
    id: '1',
    name: 'Category',
    value: 'category',
  },
  {
    id: '2',
    name: 'Brand',
    value: 'brand',
  },
  {
    id: '3',
    name: 'Contact',
    value: 'contact',
  },
  {
    id: '4',
    name: 'FAQâ€™s',
    value: 'frequently-asked-question',
  },
]
