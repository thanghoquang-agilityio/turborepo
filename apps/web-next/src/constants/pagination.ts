export const PAGE_SIZE_DEFAULT = 12
export const PAGE_DEFAULT = 1
