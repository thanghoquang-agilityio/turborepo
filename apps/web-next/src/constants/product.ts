export const STAR_DEFAULT = 5
export const COMMENTS_DEFAULT = 121
