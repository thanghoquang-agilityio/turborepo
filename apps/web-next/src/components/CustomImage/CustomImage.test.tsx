import React from 'react'

import { fireEvent, render } from '@testing-library/react'

// Constants
import { SRC_IMAGE_NOT_AVAILABLE } from '@/constants'

// Components
import { CustomImage } from '..'

// Mock next/image component
jest.mock('next/image', () => {
  return jest.fn((props) => {
    // eslint-disable-next-line @next/next/no-img-element
    return <img src={props.src} alt={props.alt} onError={props.onError} />
  })
})

// Mock useState
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}))

describe('CustomImage Component', () => {
  const props = {
    width: 100,
    height: 100,
    className: 'mock-class',
    alt: 'Mock Alt',
    src: 'mock.src',
  }

  const customImageComponent = () => {
    return render(<CustomImage {...props} />)
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('Match snapshot', () => {
    const useStateSpy = jest.spyOn(React, 'useState')

    useStateSpy.mockReturnValue([true, jest.fn()])

    const { container } = render(
      <CustomImage {...props} src={SRC_IMAGE_NOT_AVAILABLE} />
    )

    expect(container).toMatchSnapshot()
  })

  it('Sets fallback source on error', () => {
    const setState = jest.fn()
    const useStateSpy = jest.spyOn(React, 'useState')

    useStateSpy.mockReturnValue([false, setState])

    const { getByAltText } = customImageComponent()
    const image = getByAltText(props.alt)

    fireEvent.error(image)

    expect(setState).toHaveBeenCalledWith(true)
  })
})
