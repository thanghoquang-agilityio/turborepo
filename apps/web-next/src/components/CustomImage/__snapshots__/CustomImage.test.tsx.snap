// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomImage Component Match snapshot 1`] = `
<div>
  <img
    alt="UnavailableImage"
    src="/images/image-not-available.webp"
  />
</div>
`;
